### PR TITLE
Add support for addminingaddr, delminingaddr, and setminingaddrs

### PR DESCRIPTION
**Pull request by Roasbeef**
This commit adds support for the three new RPC's which were added in btcsuite/btcd#845.

### CODE REVIEW

File: extensions.go [hunk 1]
```diff
@@ -348,3 +348,114 @@ func (c *Client) SessionAsync() FutureSessionResult {
 func (c *Client) Session() (*btcjson.SessionResult, error) {
 	return c.SessionAsync().Receive()
 }
+
+// FutureSetGenerateResult is a future promise to deliver the result of a
```
Comment (davecgh) [inline extensions.go]: s/FutureSetGenerateResult /FutureAddMiningAddrResult/

File: extensions.go [hunk 2]
```diff
@@ -348,3 +348,114 @@ func (c *Client) SessionAsync() FutureSessionResult {
...
+// ListMiningAddrs returns a list of all the currently active mining addresses.
+//
+// NOTE: This is a btcsuite extension.
+func (c *Client) ListMiningAddrs() ([]string, error) {
+	return c.ListMiningAddrsAsync().Receive()
+}
+
+// FutureDelMiningAddrsResult is a future promise to deliver the result of a
+// DelMiningAddr RPC invocation (or an applicable error).
+type FutureDelMiningAddrsResult chan *response
```
Comment (davecgh) [inline extensions.go]: Along the same lines as the related PRs, would you mind defining all of these delete mining result funcs before the list addresses ones so they're consistently defined in the same order (alphabetical) as btcd/btcutil/btcrpcclient.

File: extensions.go [hunk 3]
```diff
@@ -348,3 +348,114 @@ func (c *Client) SessionAsync() FutureSessionResult {
...
+
+// Receive waits for the response promised by the future and returns true if
```
Comment (davecgh) [inline extensions.go]: Looks like this was copied but not updated.

**Review by davecgh — CHANGES_REQUESTED** [review]

File: extensions.go (hunk 3 above)
Comment (Roasbeef) [inline extensions.go]: Fixed.

File: extensions.go (hunk 2 above)
Comment (Roasbeef) [inline extensions.go]: Fixed.

File: extensions.go (hunk 1 above)
Comment (Roasbeef) [inline extensions.go]: Fixed.

**Review by davecgh — APPROVED** [review]
OK once btcsuite/btcd#845 is also merged.

**Comment by davecgh** [comment]
The rpcclient is now part of the main btcd repository.  Going to close this.  Please reopen it there.

**Pull request closed without merging**